internal/fromproto6/arguments_data.go

```diff
@@ -159,6 +160,289 @@ func ArgumentsData(ctx context.Context, arguments []*tfprotov6.DynamicValue, def
 			}
 		}
 
+		switch parameterWithValidators := parameter.(type) {
+		case function.ParameterWithBoolValidators:
+			for _, functionValidator := range parameterWithValidators.BoolValidators() {
+				boolVal, ok := attrValue.(types.Bool)
```

Comment:
Should we be using an assertion for `attrValue.(basetypes.BoolValuable)` here to accomodate the possibility that the validators have been declared on a custom type? If so, similar question for further `attrValue` type assertions in this `switch` block. 

If the `Valuable` interfaces are to be used we'll need to obtain the `BoolValue` to pass into the `BoolRequest`:

```go
bVal, diags := boolVal.ToBoolValue(ctx)
...
```